tests/test_interface.py

```diff
@@ -399,6 +415,28 @@ def test_interface_delete_payment_with_invalid_id(self, mock_db):
         self.assertNotEqual(payment_after_deletion.id,invalid_id)
         #TODO - next sprint : do get, check count, should be same
 
+#valid PUT data
+    @mock.patch.object(PaymentService,'is_valid_put',return_value = True)
+    @mock.patch.object(app_db, 'session')
+    @mock.patch('app.db.models.Payment')
+    #@mock.patch.object(Payment, 'deserialize_put')
+    #@mock.patch.object(Payment, 'serialize', return_value=PUT_CREDIT_RETURN)
+    def test_interface_update_with_valid_put_data(self,mock_isvalid,mock_db,mock_P):
+        print 'in here'
```

Comment:
here's a working version of this test:

couple important notes:
your sample data, while not the reason for the failure, was incorrect. the field `is_removed` is not returned by serialize.

the sample data `PUT_CREDIT_RETURN` is not an accurate representation of data passed as `payment_replacement`. The user won't know about fields like `charge_history`, `is_default`, `is_removed` so those would need to be updated internally.

serialize is actually called twice (as param in is_valid and at the end of the function)

i think the problem with this code was including the `get(1)`. It works without specifying the params because (i think) we're specify the return result of the `get` function as opposed to the return value of whatever `get(1)` returns. make sense?

```
    @mock.patch.object(PaymentService, 'is_valid_put', return_value=True)
    @mock.patch.object(app_db, 'session')
    @mock.patch('app.db.models.Payment')
    def test_interface_update_with_valid_put_data(self, mock_P, mock_db, mock_is_valid):
        data = PUT_CREDIT

        mock_db.query(Payment).get.return_value = mock_P
        mock_P.serialize.return_value = PUT_CREDIT_RETURN
        resp = self.ps.update_payment(id, payment_replacement=data)
        mock_P.serialize.assert_called()
        mock_is_valid.assert_called_once()
        mock_P.deserialize_put.assert_called_once()
        mock_db.commit.assert_called_once()
        self.assertEqual(resp['nickname'],'favcredit')
```